rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_storage.cpp

```diff
@@ -214,21 +223,20 @@ std::unique_ptr<rosbag2_storage::BagMetadata> SqliteStorage::load_metadata(const
   }
 }
 
-bool SqliteStorage::database_exists(const std::string & uri)
+bool SqliteStorage::is_read_only(const rosbag2_storage::storage_interfaces::IOFlag & io_flag) const
 {
-  std::ifstream database(uri);
-  return database.good();
+  return io_flag == rosbag2_storage::storage_interfaces::IOFlag::READ_ONLY;
 }
```

Comment:
How is that change related? Why introducing `is_read_only` here?

Reply:
That function already existed. I think the reason why the diff lists it as a change is because I moved `database_exists` to the top of the file since its now a free function.